kratos/geometries/geometry_shape_function_container.h

```diff
@@ -0,0 +1,332 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//                   Pooyan Dadvand
+//
+//
+
+#if !defined(KRATOS_GEOMETRY_SHAPE_FUNCTION_CONTAINER_H_INCLUDED )
+#define  KRATOS_GEOMETRY_SHAPE_FUNCTION_CONTAINER_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+
+namespace Kratos
+{
+
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/** GeometryShapeFunctionContainer.
+    Holds the evaluated values for the shape functions and its derivatives.
+    Needs to be templated by the IntegrationMethod to import the
+    IntegrationMethod enum which is defined in the GeometryData.
+*/
+template<typename TIntegrationMethodType>
+class GeometryShapeFunctionContainer
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of GeometryShapeFunctionContainer
+    KRATOS_CLASS_POINTER_DEFINITION( GeometryShapeFunctionContainer );
+
+    using IntegrationMethod = TIntegrationMethodType;
+
+    typedef std::size_t IndexType;
+    typedef std::size_t SizeType;
+
+    typedef IntegrationPoint<3> IntegrationPointType;
+    typedef std::vector<IntegrationPointType> IntegrationPointsArrayType;
+    typedef std::array<IntegrationPointsArrayType, IntegrationMethod::NumberOfIntegrationMethods> IntegrationPointsContainerType;
+
+
+    typedef std::array<Matrix, IntegrationMethod::NumberOfIntegrationMethods> ShapeFunctionsValuesContainerType;
+    typedef std::array<DenseVector<Matrix>, IntegrationMethod::NumberOfIntegrationMethods> ShapeFunctionsLocalGradientsContainerType;
+
+    typedef DenseVector<Matrix> ShapeFunctionsGradientsType;
+    typedef DenseVector<Matrix> ShapeFunctionsSecondDerivativesType;
+    typedef DenseVector<DenseVector<Matrix> > ShapeFunctionsThirdDerivativesType;
```

Comment:
This is not coherent with the new design. Isn't it?

Reply:
It is removed in #5424